From 7fb970f0b08428dea6d964d265bd16d1ed1e36c1 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@linaro.org>
Date: Wed, 6 Apr 2016 15:58:31 +0200
Subject: [PATCH 4/7] STM: iec61937: set iec958 boolean before use it

During prepare alsa->iec958  is tested before set.
Effect is that alsa device is not opened in iec958 mode,
so AES value are not transmitted to alsa driver.

Change-Id: I9c3a2d29003ddebe875a0995801208b09d51d774
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

diff --git a/ext/alsa/gstalsasink.c b/ext/alsa/gstalsasink.c
index eed4d6c..b6ff856 100644
--- a/ext/alsa/gstalsasink.c
+++ b/ext/alsa/gstalsasink.c
@@ -878,6 +878,9 @@ gst_alsasink_prepare (GstAudioSink * asink, GstAudioRingBufferSpec * spec)
 
   alsa = GST_ALSA_SINK (asink);
 
+  if (!alsasink_parse_spec (alsa, spec))
+    goto spec_parse;
+
   if (alsa->iec958) {
     snd_pcm_close (alsa->handle);
     alsa->handle = gst_alsa_open_iec958_pcm (GST_OBJECT (alsa), alsa->device);
@@ -886,9 +889,6 @@ gst_alsasink_prepare (GstAudioSink * asink, GstAudioRingBufferSpec * spec)
     }
   }
 
-  if (!alsasink_parse_spec (alsa, spec))
-    goto spec_parse;
-
   CHECK (set_hwparams (alsa), hw_params_failed);
   CHECK (set_swparams (alsa), sw_params_failed);
 
-- 
2.7.4

